dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp3ProtocolTest.java

```
@@ -87,50 +100,70 @@ void testDemoProtocol() throws Exception {
         MockStreamObserver outboundMessageSubscriber2 = new MockStreamObserver();
         StreamObserver<String> inboundMessageObserver = greeterProxy.bidirectionalStream(outboundMessageSubscriber2);
         inboundMessageObserver.onNext(REQUEST_MSG);
         inboundMessageObserver.onCompleted();
         outboundMessageSubscriber2.getLatch().await(3000, TimeUnit.MILLISECONDS);
-        // verify client
+
+        // Verify client
         Assertions.assertEquals(IGreeter.SERVER_MSG, outboundMessageSubscriber2.getOnNextData());
         Assertions.assertTrue(outboundMessageSubscriber2.isOnCompleted());
-        // verify server
+
+        // Verify server
         MockStreamObserver serverOutboundMessageSubscriber = (MockStreamObserver) serviceImpl.getMockStreamObserver();
         serverOutboundMessageSubscriber.getLatch().await(1000, TimeUnit.MILLISECONDS);
         Assertions.assertEquals(REQUEST_MSG, serverOutboundMessageSubscriber.getOnNextData());
         Assertions.assertTrue(serverOutboundMessageSubscriber.isOnCompleted());
 
         export.unexport();
         protocol.destroy();
-        // resource recycle.
         serviceRepository.destroy();
-        System.out.println("serviceRepository destroyed");
+        System.out.println("ServiceRepository destroyed at end of test");
+    }
+
+    @AfterEach
+    void cleanup() {
+        if (applicationModel != null) {
+            try {
+                DefaultRequestMappingRegistry registry = applicationModel
+                        .getFrameworkModel()
+                        .getBeanFactory()
+                        .getBean(DefaultRequestMappingRegistry.class);
+                if (registry != null) {
+                    registry.destroy();
+                    System.out.println("DefaultRequestMappingRegistry destroyed after test");
+                }
+            } catch (Exception e) {
+                System.err.println("Cleanup error: " + e.getMessage());
+            }
```

Exception should be thrown instead of just logging the error.